svelte/src/lib/components/FileExplorer/Node.svelte

```diff
@@ -1,295 +1,338 @@
-            }
-        }
-
-        addMode = false;
-    };
-
-    const cancelRename = () => {
-        appStateDispatch({
-            type: AppActionType.UpdateFile,
-            nodePath: nodePath,
-            value: { editMode: false },
-        });
-    };
-
-    const submitRename = async (e: any) => {
-        const { value: newName } = e.detail;
-
-        // if not change, just exit edit mode
-        if (label === newName) {
-            appStateDispatch({
-                type: AppActionType.UpdateFile,
-                nodePath: nodePath,
-                value: { editMode: false },
-            });
-            return;
-        }
-
-        // if change node name, invoke rename action
-        let parentFilePath = extractPath(fileSystemPath, uiContext.pathSeparator).parentPath;
-        let newFileSystemPath = parentFilePath + uiContext.pathSeparator + newName;
-        const response = await fileSystem.rename(fileSystemPath, newFileSystemPath);
-        if (response.isSuccess) {
-            // leave watch handler update tree (remove old node, add new node)
-
-            // update selectedPath
-            const parentNodePath = NodeInfo.extractPath(nodePath).parentPath;
-            let newNodePath = NodeInfo.combinePath(parentNodePath, newName);
-            appStateDispatch({
-                type: AppActionType.SelectFile,
-                nodePath: newNodePath,
-            });
-
-            // let's update related tab if any
-            const tabIndex = $appState.tabs.findIndex((t) => t.id === nodePath);
-            if (tabIndex < 0) {
-                return;
-            }
-
-            let tabToUpdate = {
-                id: newNodePath,
-                title: newName,
-                tooltip: newFileSystemPath,
-                content: {
-                    data: {
-                        folderPath: parentFilePath,
-                        fileName: newName,
-                    },
-                },
-            } as IGenericTab<IFileInfo>;
-            appStateDispatch({
-                type: AppActionType.UpdateTab,
-                tabIndex: tabIndex,
-                value: tabToUpdate,
-            });
-        } else {
-            // false, let's show error if any
-            if (response.errorMessage) {
-                Notify.error(response.errorMessage);
-            }
-
-            // then exit edit mode
-            appStateDispatch({
-                type: AppActionType.UpdateFile,
-                nodePath: nodePath,
-                value: { editMode: false },
-            });
-        }
-    };
+    }
+
+    console.log("generateProjectMetadata", meta);
+    console.log("testCase", testCase);
+    console.log("relatedTestSuiteRelPaths", relatedTestSuiteRelPaths);
+    showDeleteTestCaseDialog = true;
+  };
+
+  const handleMenuDelete = async () => {
+    if (type === FileType.TCase) {
+      await handleDeleteTestCase();
+      return;
```

Comment:
**Important**: From my view point, we should NOT let DeleteTestCaseDialog handle everything, such as:  
- Delete test case
- Modify test suite
- Close tab
- .... (may be something else in future)

With above approach, the dialog needs to handle 'delete file' logic and related stuff, such as close tab (maybe, in future, we need to do extra work as well).   Moreover, if there is any update about delete-file logic, we have to update 2 places:  FileExplorer/Node.svelte and DeleteTestCaseDialog.svelte

I would recommend:
- If that is the case we need to handle (example: file type == TCase), then:    // call -> handleTestCaseDeletion
- Perform additional logic, such as generate code metadata then check whether test case is being used by test suite,
- If no,  execute normal deletion workflow as other scenario / file
- If yes, collect info (related suite full path, test case id), then display Delete Test Case (confirmation) dialog
- If user confirm NO:  do nothing, stop
- If user confirm YES:  on the dialog,  perform test suite modification first,  then dispatch custom event  (event can be:  'delete'  or 'delete-node' with the parameter is node path (or corresponding (test-case) fileSystemPath)
- Outside dialog, at the Node.svelte,  just handle that custom event then execute normal node deletion logic  (include other logic like close tab)
- In summary,  logic of delete file should leave or centralize at Node.svelte.     custom Delete...Dialog just handle its own special logic like modify  test-suite content